lib/config/presets/index.ts

```diff
@@ -384,11 +407,41 @@ export async function resolveConfigPresets(
         val as RenovateConfig,
         baseConfig,
         ignorePresets,
-        existingPresets
+        existingPresets,
+        shallowResolve
       );
     }
   }
   logger.trace({ config: inputConfig }, 'Input config');
   logger.trace({ config }, 'Resolved config');
+  if (shallowResolve) {
+    handleExtendsArray(config, unresolvedPreserts);
+  }
   return config;
 }
+
+function handleExtendsArray(config: AllConfig, tempExtends: string[]): void {
+  // in case of shallowResolve set to true:
+  // remove resolved external presets from the extends array for the log
+  // clean duplicate presets in case of two different presets are having
+  // same values in the extends array
+  config.extends = config.extends?.filter((e) => !shouldShallowResolve(e));
+  config.extends?.push(...tempExtends);
```

Comment:
Is it intended that tempExtends is ignored if config.extends is missing? If so then perhaps just check for config.extends first and return immediately within this function?

Reply:
Yes didn't notice that.